src/ripple/shamap/impl/SHAMapSync.cpp

```diff
@@ -327,7 +323,7 @@ SHAMap::getMissingNodes(int max, SHAMapSyncFilter* filter)
     MissingNodes mn(
         max,
         filter,
-        f_.db().getDesiredAsyncReadCount(ledgerSeq_),
+        4096,   // number of async reads per pass
```

Comment:
Should this number be tunable? Is it possible that 4096 will be too much (or too little)?

Reply:
There's really no basis on which to tune it. It controls how quickly getMissingNodes switches between looking for more async reads to do and waiting for async reads to complete. There's no reason to think more is better, you just need a high enough number that the cost of waiting for reads to complete is lost in the noise. You just want to stay in the "discovering reads we need to do" phase long enough that a couple of the reads you've pended have completed to minimize the time you're doing nothing.

Reply:
:+1: